### PR TITLE
Add CDI2.0 Security Context code and tests

### LINKED ISSUE #246: CDI 2.0: SecurityServices impl
http://docs.jboss.org/weld/javadoc/3.0/weld-spi/org/jboss/weld/security/spi/SecurityServices.html has gained one more method getSecurityContext() - return current security context, which can then be propagated to different threads

### PR DESCRIPTION
**Pull request by neilgsyoung**
fixes #246

### CODE REVIEW

**Comment by neilgsyoung** [comment]
#build

**Comment by neilgsyoung** [comment]
#build

File: dev/com.ibm.ws.cdi.2.0.weld/src/com/ibm/ws/cdi/impl/CDI20SecurityContext.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import com.ibm.wsspi.threadcontext.ThreadContextDescriptor;
+
+/**
+ * The Liberty implementation of the
+ * org.jboss.weld.security.spi.SecurityContext interface
+ */
+class CDI20SecurityContext implements SecurityContext {
+    private static final TraceComponent tc = Tr.register(CDI20SecurityContext.class);
+    private ThreadContextDescriptor threadContextDesc = null;
+    private ArrayList<ThreadContext> threadContext = null;
```
Comment (Emily-Jiang) [inline dev/com.ibm.ws.cdi.2.0.weld/src/com/ibm/ws/cdi/impl/CDI20SecurityContext.java]: Looks like this variable is not needed.

File: dev/com.ibm.ws.cdi.2.0_fat/test-applications/secureAsyncEventsApp/src/secureAsyncEventsApp/web/SecureAsyncEventsServlet.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        Subject runAsSubject = null;
+
+        try {
+            runAsSubject = WSSubject.getRunAsSubject();
+        } catch (Exception e2) { // WSSecurityException
+            e2.printStackTrace();
+        }
+
+        CakeArrival newCake = new CakeArrival();
+        CompletionStage<CakeArrival> stage = cakeEvent.fireAsync(newCake);
```
Comment (Emily-Jiang) [inline dev/com.ibm.ws.cdi.2.0_fat/test-applications/secureAsyncEventsApp/src/secureAsyncEventsApp/web/SecureAsyncEventsServlet.java]: Can you also provide a executor on fireAsync?

**Comment by neilgsyoung** [comment]
#build

**Comment by neilgsyoung** [comment]
#build

**Comment by neilgsyoung** [comment]
#build

**Comment by neilgsyoung** [comment]
#build

**Comment by neilgsyoung** [comment]
#build

**Pull request merged**